hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/CallIdSequenceWithBackpressure.java

```diff
@@ -94,13 +99,20 @@ private boolean hasSpace() {
         return longs.get(INDEX_HEAD) - longs.get(INDEX_TAIL) < maxConcurrentInvocations;
     }
 
-    private void waitForSpace() throws TimeoutException {
-        long deadline = System.nanoTime() + backoffTimeoutNanos;
+    private void waitForSpace() {
+        if (backoffTimeoutNanos <= 0) {
+            throw new HazelcastOverloadException(String.format(
```

Comment:
This if check should not be needed anymore because we know that backoffTimeoutNanos is positive (you verified it in the constructor)

Reply:
OK, corrected.